numpy/f2py/f2py2e.py

```diff
@@ -318,6 +325,16 @@ def scaninputline(inputline):
     options['buildpath'] = buildpath
     options['include_paths'] = include_paths
     options.setdefault('f2cmap_file', None)
+    if not emptygen:
```

Comment:
Please don't do this, generating warnings for usage that's perfectly valid without any changes is not a good practice.

Note that this is independent from the default behavior that's chosen, this warning should not be here either way.

Reply:
Makes sense, I'll switch the default now and add a negative flag for backwards compatibility.